tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+
+            def getKey(item):
+                return item[0]
+
+            sorted_feature_names = sorted(feature_names, key=getKey)
+            sorted_tsfresh_features = sorted(TSFRESH_FEATURES, key=getKey)
+
+            for nid, nname in sorted_feature_names:
+                new_entry = None
+                if int(nid) > max_known_id:
+                    new_entry = '    [%s, %s]' % (str(nnid), str(nname))
+                    fail_msg = '''
+
+###    NOTICE    ###
+
+If you adding/testing new features, you can create a new baseline from: %s
+
+Added the new baseline (Python 2 and 3 baselines ARE different) as either:
+Local path: %s/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+Repo path: tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+
+In the tsfresh_feature_names.py change the following:
+Local path: %s/tsfresh_feature_names.py
+Repo path: tests/baseline/tsfresh_feature_names.py
+
+- TSFRESH_BASELINE_VERSION to your <NEW_VERSION>
+- Add the new feature/s name to TSFRESH_FEATURES with a incremented id value
+
+New entry for tsfresh_feature_names.py:
+%s
+''' % (t_fname_out_fail, baseline_dir, baseline_dir, str(new_entry))
+                    if new_entry:
+                        shutil.move(t_fname_out, t_fname_out_fail)
+                    self.assertEqual(new_entry, None, msg=fail_msg)
+
+                for oid, oname in sorted_feature_names:
+                    if int(oid) == int(nid):
+                        if str(oname) != str(nname):
+                            soid = str(int(oid))
+                            snid = str(int(nid))
+                            soname = str(oname)
+                            snname = str(oname)
+                            fail_msg = '''
+
+###    ERROR    ###
+
+A baseline feature name for id %s has changed:
+    [%s, %s]
+Calculated feature name for id %s was:
+    [%s, %s]
+
+If you adding/testing new features, you can create a new baseline from: %s
+
+Added the new baseline (Python 2 and 3 baselines ARE different) as either:
+Local path: %s/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+Repo path: tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
+
+In the tsfresh_feature_names.py change the following:
+Local path: %s/tsfresh_feature_names.py
+Repo path: tests/baseline/tsfresh_feature_names.py
+
+- TSFRESH_BASELINE_VERSION to your <NEW_VERSION>
+- Add the new feature/s name to TSFRESH_FEATURES with a incremented id value
+
+''' % (soid, soid, soname, soid, soid, snname, t_fname_out, baseline_dir, baseline_dir)
+                            if soname != snname:
+                                shutil.move(t_fname_out, t_fname_out_fail)
+                            self.assertEqual(soname, snname, msg=fail_msg)
+            feature_names_match = True
+        else:
+            feature_names_match = False
+            shutil.move(t_fname_out, t_fname_out_fail)
```

Comment:
And if it was just a feature addition, obviously I added this to the transposed csv not to tsfresh :)

```python
>       self.assertTrue(features_equal, msg=fail_msg)
E       AssertionError:
E
E       ###    NOTICE    ###
E
E       If you adding/testing new features, you can create a new baseline from: /opt/python_virtualenv/projects/tsfresh-py2712/apps/tsfresh_test_tsfresh/tsfresh/tsfresh/examples/data/test_tsfresh_baseline_dataset/tsfresh-unknown-version.py2.data.json.features.transposed.csv
E
E       Added the new baseline (Python 2 and 3 baselines ARE different) as either:
E       Local path: /opt/python_virtualenv/projects/tsfresh-py2712/apps/tsfresh_test_tsfresh/tsfresh/tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
E       Repo path: tests/baseline/tsfresh-<NEW_VERSION>.py{2,3}.data.json.features.transposed.csv
E
E       In the tsfresh_feature_names.py change the following:
E       Local path: /opt/python_virtualenv/projects/tsfresh-py2712/apps/tsfresh_test_tsfresh/tsfresh/tests/baseline/tsfresh_feature_names.py
E       Repo path: tests/baseline/tsfresh_feature_names.py
E
E       - TSFRESH_BASELINE_VERSION to your <NEW_VERSION>
E       - Add the new feature/s name to TSFRESH_FEATURES with a incremented id value
E
E       NOT in baseline   :: []
E
E       NOT in calculated :: [['value__I_ADDED_A_NEW_FEATURE', '123.0']]
```